### ISSUE TITLE
Open Sans, Roboto and Roboto Slab NAME table issues in WinWord

### ISSUE THREAD

**Issue by davelab6** [google/fonts#1618]
A user wrote to me privately that Open Sans, Roboto and Roboto Slab all have NAME table issues in Windows Word. 

Open Sans when installed looks like this to Word:

![unnamed-1](https://user-images.githubusercontent.com/261579/42766020-5a89c592-88e7-11e8-8661-0fb772e1e918.jpg)

So far so good.

However, the SemiBold Italic style does not render, instead the SemiBold is auto-slanted (eg the 'a' is not single story)

![image007](https://user-images.githubusercontent.com/261579/42766030-5ce027d2-88e7-11e8-8053-5c692879a846.jpg)

Similarly, Roboto Slab has 4 styles (Regular + Bold, then Thin and then Light) but only two appear in the Word picker list:

![unnamed](https://user-images.githubusercontent.com/261579/42766022-5aa02d96-88e7-11e8-8f6f-ed381a16be9c.jpg)

So when a document is made with them in Google Docs and exported as docx, it appears to fall back to Roboto:

![image005](https://user-images.githubusercontent.com/261579/42766026-5bf8075e-88e7-11e8-991c-6b2643792acd.jpg)

I am very keen to see that Font Bakery is checking this stuff and telling people exactly what to do to fix it, and then we ensure everything in the API is fixed. :)

**Comment by felipesanches** [comment]
OK. I'll investigate this.

**Comment by laerm0** [comment]
Hi –

I can't reproduce this with Open Sans (using the files currently in the repo, not my new files) or Roboto but I can with Roboto Slab.

I'll fix Roboto Slab. I'm surprised nobody has ever noticed this before!

![screen shot 2018-07-17 at 10 56 19](https://user-images.githubusercontent.com/10780161/42829497-bcf0c69c-89b7-11e8-8db1-291832801e5f.png)
![screen shot 2018-07-17 at 10 48 51](https://user-images.githubusercontent.com/10780161/42829498-bd0bdef0-89b7-11e8-81d8-9ee7b1ea24d3.png)

**Comment by laerm0** [comment]
It looks like @mjlagattuta has just spiffed up Roboto Slab's masters for VF lifestyle so it looks like getting the slab live on site will fix this (and improve other stuff).

**Comment by felipesanches** [comment]
@laerm0 sent me two sets of good and bad fonts. Here's a visual reference of how the name table entries went bad on Roboto:

**Roboto Thin (good)** versus **Roboto Thin (bad)** 
![screenshot at 2018-07-17 14 25 12](https://user-images.githubusercontent.com/213676/42835117-ae220fba-89e7-11e8-9ef1-f190e2ecd0bc.png)

**Roboto Light (good)** versus **Roboto Light (bad)** 
![screenshot at 2018-07-17 14 25 58](https://user-images.githubusercontent.com/213676/42835116-adfe6376-89e7-11e8-8337-0c3d209a4305.png)

**Roboto Regular (good)** versus **Roboto Regular (bad)** 
![screenshot at 2018-07-17 14 22 53](https://user-images.githubusercontent.com/213676/42835120-ae651724-89e7-11e8-80d7-e55601e3b711.png)

**Roboto Bold (good)** versus **Roboto Bold (bad)** 
![screenshot at 2018-07-17 14 23 41](https://user-images.githubusercontent.com/213676/42835119-ae436ab6-89e7-11e8-8796-51a6acea2c23.png)

**Comment by davelab6** [comment]
@laerm0 

> I can't reproduce this with Open Sans (using the files currently in the repo, not my new files)

Did you mean, "using the files currently in the repo, OR my new files"

**Comment by davelab6** [comment]
Also, what versions of win & word did you test? :)

**Comment by laerm0** [comment]
Sorry, using the files currently in the repo.

I used Win 10 Pro with Office 365.

**Comment by laerm0** [comment]
The files currently in the repo were what I tested, but I just downloaded what's being served again and it's the same as what I saw before: Open Sans and Roboto fine, Roboto Slab not.

**Comment by mjlagattuta** [comment]
In the Roboto Slab VF (which is still in beta) I resolved this via manual edits to the STAT table, which Windows Word seems to rely on. I believe these values need to correspond to the final scale of an axis whether scaling from source file stem weights to CSS weight values and/or mapping via an avar table. The only style not present is "Roboto Slab Beta Bold" which can be accessed via turning on the bold option on the regular style.

<img width="263" alt="screen shot 2018-07-23 at 4 29 14 pm" src="https://user-images.githubusercontent.com/35570153/43101401-523e286a-8e96-11e8-9dc7-3fb973360d75.png">

Markhazi Text for example renders in Arial in Windows Word because its STAT values correspond to its source file values for weight rather than its scaled values 400–700 and since the source file values are not even on the final range the font does not display at all

**Comment by felipesanches** [comment]
Do we have a succinct description of the rules for good and bad name table entries relevant to the issues discussed here?

**Comment by felipesanches** [comment]
I need someone to take a look at this issue and provide some further feedback, please.

**Comment by davelab6** [comment]
> Do we have a succinct description of the rules for good and bad name table entries relevant to the issues discussed here?

No. Someone must investigate systematically and write those rules, and then write code that checks them. 

We have fonts that work, and fonts that do not work, so using fontdiffenator (or even just comparing the TTX output of the `OS/2`, `name` and `post` tables, by hand) should reveal what is going on.

**Comment by m4rc1e** [comment]
@mjlagattuta Thanks for the write up. We have this info has a [spreadsheet](https://docs.google.com/spreadsheets/d/1ckHigO7kRxbm9ZGVQwJ6QJG_HjV_l_IRWJ_xeWnTSBg/edit#gid=0)

Has anyone been able to verify what this user has found?

**Comment by mjlagattuta** [comment]
Same as @laerm0 I cannot replicate this for Open Sans and Roboto but I can for Roboto Slab. I just ran the four fonts through FontBakery. Checks 157 and 158 yielded `FAIL` for both Roboto Slab Light and Roboto Slab Thin and suggested the correct name string that was expected, so it would seem FontBakery is successfully detecting these `name` table errors

**Comment by felipesanches** [comment]
The implementation of `com.google.fonts/check/156` is correct indeed, but its code-test was not, so I have just fixed it at https://github.com/felipesanches/fontbakery/commit/4fa6cefddc29418bc842fab57d1615ebaef3a8de

**Comment by felipesanches** [comment]
I am reviewing the other 2 checks. They seem to be correctly implemented as well, but lack code-tests, so I am working on it right now.

**Comment by felipesanches** [comment]
https://github.com/felipesanches/fontbakery/commit/22fea7d58098270c798840186021ac2afe1a92d3 fixes checks 157 and 158. It also implements code-test for check/157.

**Comment by felipesanches** [comment]
@mjlagattuta, at https://github.com/google/fonts/issues/1618#issuecomment-425948569 you said:
> Check 158 then makes sure that the entries at nameID 2 can only be RIBBI.

But reading the current implementation of check/158 I see:

```python
  for name in ttFont['name'].names:
    if name.nameID == NameID.FONT_SUBFAMILY_NAME:
      if style_with_spaces not in STYLE_NAMES:
        yield FAIL, ("Style name '{}' inferred from filename"
                     " is not canonical."
                     " Valid options are: {}").format(style_with_spaces,
                                                      STYLE_NAMES)
        failed = True
        continue
```

and at **constants.py** we have `STYLE_NAMES` declared as:

```python
STYLE_NAMES = ["Thin",
               "ExtraLight",
               "Light",
               "Regular",
               "Medium",
               "SemiBold",
               "Bold",
               "ExtraBold",
               "Black",
               "Thin Italic",
               "ExtraLight Italic",
               "Light Italic",
               "Italic",
               "Medium Italic",
               "SemiBold Italic",
               "Bold Italic",
               "ExtraBold Italic",
               "Black Italic"]
```

So that's defnitely not RIBBI-only...

Can you review that comment, please?

**Comment by mjlagattuta** [comment]
That portion of code only checks to make sure that the style name is canonical.

I believe my comment on the check was in reference to this snippet that is just below the one you referenced:
```python
      if name.platformID == PlatformID.MACINTOSH:
        expected_value = style_with_spaces

      elif name.platformID == PlatformID.WINDOWS:
        if style_with_spaces in ["Bold", "Bold Italic"]:
          expected_value = style_with_spaces
        else:
          if "Italic" in style:
            expected_value = "Italic"
          else:
            expected_value = "Regular"
      else:
        yield FAIL, ("Font should not have a "
                     "{} entry!").format(name_entry_id(name))
        failed = True
```
After checking that the style name is canonical then this checks that anything that is not Bold, Bold Italic, or Italic should be Regular

I think this is correct for checking static fonts, but for Variable fonts, it is not, since we may want to differentiate between the font's default outlines and the regular instance (if those two things are not the same). As an example, Comfortaa VF, has a font subfamily name of "Light" in reference to its default outlines, which ensures that the instances in dropdowns on Mac do not display two "Regulars".

**Comment by felipesanches** [comment]
The test case describes by example the expected behaviour:
https://github.com/googlefonts/fontbakery/blob/94bffc3a3ce0562329df4d1635268bf10ec5ba9e/tests/specifications/googlefonts_test.py#L2080-L2104

**Comment by Spherikal** [comment]
Did this ever get a solution? I still can't use Roboto Slab in Word...

**Comment by julieyunt** [comment]
Also can't use **Roboto Slab - regular** in Word (on a PC)  It reverts to **Roboto Slab - Light**. 
How do we get this fixed.
